product_contract/models/product_template.py

```diff
@@ -48,7 +50,7 @@ def _change_is_contract(self):
         product from contract
         """
         if not self.is_contract:
-            self.contract_template_id = False
+            self.property_contract_template_id = False
```

Comment:
This is not valid in a multi-company context. Instead, you must overwrite `write` method and if `is_contract` is False, remove the value for all the companies.